sdk/src/main/java/com/bugsnag/android/JsonStream.java

```diff
@@ -49,6 +49,10 @@ public void value(@Nullable Streamable streamable) throws IOException {
      * Writes a File (its content) into the stream
      */
     public void value(@NonNull File file) throws IOException {
+        if (file == null || file.length() <= 0) {
```

Comment:
Do you need the `null` check if `@NonNull` is used above?

Reply:
`@NonNull` doesn't enforce anything in Java unfortunately. The IDE would highlight any null parameters with a warning, but developers could choose to ignore this.

Reply:
Ah, fair enough.